From a1090af83a7d7f96bcbba07c804b7e1873fabde9 Mon Sep 17 00:00:00 2001
From: chdugw_auto <jkulkarni@maxlinear.com>
Date: Fri, 20 Aug 2021 12:26:04 +0300
Subject: [PATCH] Merge pull request #3487 in SW_UGW/linux from
 bugfix/UGW_SW-61381-jumbo-test-cbm-buffer-leak-fix to master

* commit '81662387c5b08e9abf619258cec2e2f6ea330a00':
  UGW_SW-61381: cqm grx500: cbm_buffer_free_grx500() with virtual pointer to fix cbm buffer leak
---
 drivers/net/ethernet/lantiq/cqm/grx500/cbm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c b/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c
index d8e0ca5be9a0..be55037c4edb 100644
--- a/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c
+++ b/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c
@@ -2219,7 +2219,7 @@ static int do_dq_cbm_poll(struct napi_struct *napi, int budget)
 				LOGF_KLOG_DEBUG("%s: rx to datapath library\n", __func__);
 			} else {
 				LOGF_KLOG_ERROR("%s:failure in allocating skb, free cbm data pointer\r\n", __func__);
-				cbm_buffer_free_grx500(smp_processor_id(), desc_list->desc.desc2, 0);
+				cbm_buffer_free_grx500(smp_processor_id(), (void *)data_ptr, 0);
 			}
 		} else {
 				tasklet_loop = (tasklet_loop >= 2) ? 0 : tasklet_loop + 1;
@@ -2346,7 +2346,7 @@ static void do_cbm_tasklet(unsigned long cpu)
 				LOGF_KLOG_DEBUG("%s: rx to datapath library\n", __func__);
 			} else {
 				LOGF_KLOG_ERROR("%s:failure in allocating skb, free cbm data pointer\r\n", __func__);
-				cbm_buffer_free_grx500(smp_processor_id(), (void *)desc_list->desc.desc2, 0);
+				cbm_buffer_free_grx500(smp_processor_id(), (void *)data_ptr, 0);
 			}
 			desc_list = &g_cbm_dlist[cpu][++j];
 		}
